src/mongocxx/model/write.cpp

```diff
@@ -57,36 +90,34 @@ void write::destroy_member() {
         case write_type::k_replace_one:
             _replace_one.~replace_one();
             break;
-        case write_type::k_uninitialized:
-            break;
     }
-
-    _type = write_type::k_uninitialized;
 }
 
 write& write::operator=(write&& rhs) noexcept {
+    if (this == &rhs) {
```

Comment:
Do we actually need to guard against self move assign? I consider `a = std::move(a)` to be a bug. However, I would like `std::swap(a, a)` to work.